avro4s-core/src/main/scala/com/sksamuel/avro4s/AvroInputStream.scala

```diff
@@ -15,11 +15,11 @@ trait AvroInputStream[T] {
   def tryIterator(): Iterator[Try[T]]
 }
 
-class AvroBinaryInputStream[T](in: InputStream)
+class AvroBinaryInputStream[T](in: InputStream, writerSchema: org.apache.avro.Schema = null)
```

Comment:
Maybe we're better off breaking the API and having the input stream always need a writer, of which it can be provided via an implicit on an apply method?